src/UI/class-admin-columns-analytics.php

```diff
@@ -0,0 +1,248 @@
+
+use const Parsely\Utils\WP_MAX_POSTS_PER_PAGE;
+use const Parsely\Utils\DATE_TIME_UTC_FORMAT;
+
+/**
+ * Shows `Parse.ly Stats` on Admin Columns
+ *
+ * @since 3.7.0
+ *
+ * @phpstan-import-type Analytics_Post_API_Params from Analytics_Posts_API
+ * @phpstan-import-type Analytics_Post from Analytics_Posts_API
+ * @phpstan-import-type Analytics_Post_Metrics from Analytics_Posts_API
+ */
+final class Admin_Columns_Analytics {
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Internal Variable.
+	 *
+	 * @var array<string, Analytics_Post_Metrics>
+	 */
+	private $parsely_stats_map = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Registers action and filter hook callbacks.
+	 *
+	 * @since 3.7.0
+	 */
+	public function run(): void {
+		if ( $this->parsely->site_id_is_set() && $this->parsely->api_secret_is_set() ) {
+			add_filter( 'the_posts', array( $this, 'set_parsely_stats' ) );
+			// TODO: Add columns for pages and custom post types.
+			add_filter( 'manage_posts_columns', array( $this, 'add_parsely_stats_column_on_list_view' ) );
+			add_action( 'manage_posts_custom_column', array( $this, 'show_parsely_stats' ) );
+		}
+	}
+
+	/**
+	 * Calculate Min and Max Publish date from all the found posts.
+	 *
+	 *  @param WP_Post[] $posts Array of post objects.
+	 *
+	 * @return WP_Post[]
+	 */
+	public function set_parsely_stats( array $posts ): array {
+		$post_type = is_null( get_current_screen() ) ? '' : get_current_screen()->post_type;
+		// TODO: Add support for columns of pages and custom post types.
+		if ( 'post' !== $post_type ) {
+			return $posts;
+		}
+
+		$date_params = $this->get_publish_date_params_for_analytics_api( $posts );
+		if ( is_null( $date_params ) ) {
+			return $posts;
+		}
+
+		$analytics_api = new Analytics_Posts_API( $this->parsely );
+		$response      = $analytics_api->get_post_analytics(
+			array(
+				'pub_date_start' => $date_params['pub_date_start'] ?? '',
+				'pub_date_end'   => $date_params['pub_date_end'] ?? '',
+				'limit'          => WP_MAX_POSTS_PER_PAGE,
+				'sort'           => 'avg_engaged',
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			// TODO: handle error.
+			return $posts;
+		}
+
+		foreach ( $response as $analytics_post ) {
+			$key = $this->get_unique_stats_key_from_analytics( $analytics_post );
+
+			if ( '' !== $key && isset( $analytics_post['metrics'] ) ) {
+				$this->parsely_stats_map[ $key ] = $analytics_post['metrics'];
+			}
+		}
+
+		return $posts;
+	}
+
+	/**
+	 * Adds the `Parse.ly Stats` on admin columns.
+	 *
+	 * @param array<string, string> $columns Columns array which contain keys and labels.
+	 *
+	 * @return array<string, string>
+	 */
+	public function add_parsely_stats_column_on_list_view( array $columns ): array {
+		$columns['parsely-stats'] = 'Parse.ly Stats';
```

Comment:
I think that this column is to be named "Parse.ly Content Helper" or "Content Helper". You can verify this with @danielabloch.

Reply:
Picked this from Figma designs and the name looks good to me, will take feedback from design or product on this.